src/test/java/com/artipie/http/TrafficMetricSliceTest.java

```
@@ -0,0 +1,108 @@
+
+    /**
+     * Response metrics key.
+     */
+    private static final String KEY_RSP = "response.body.size";
+
+    @Test
+    void updatesMetrics() throws Exception {
+        final Metrics metrics = new InMemoryMetrics();
+        final int size = 100;
+        new TrafficMetricSlice((line, head, body) -> new RsWithBody(body), metrics)
+            .response(
+                new RequestLine(RqMethod.GET, "/foo").toString(),
+                Collections.emptyList(),
+                Flowable.just(ByteBuffer.wrap(new byte[size]))
+            )
+            .send(
+                (status, headers, body) ->
+                    Flowable.fromPublisher(body).ignoreElements().to(CompletableInterop.await())
+            ).toCompletableFuture().get();
+        final AtomicLong request = new AtomicLong();
+        final AtomicLong response = new AtomicLong();
+        metrics.publish(
+            // @checkstyle AnonInnerLengthCheck (22 lines)
+            new MetricsOutput() {
+                @Override
+                public void counters(final Map<String, Long> data) {
+                    if (data.containsKey(TrafficMetricSliceTest.KEY_REQ)) {
+                        request.addAndGet(data.get(TrafficMetricSliceTest.KEY_REQ));
+                    }
+                    if (data.containsKey(TrafficMetricSliceTest.KEY_RSP)) {
+                        response.addAndGet(data.get(TrafficMetricSliceTest.KEY_RSP));
+                    }
+                }
+
+                @Override
+                public void gauges(final Map<String, Long> data) {
+                    if (data.containsKey(TrafficMetricSliceTest.KEY_REQ)) {
+                        request.set(data.get(TrafficMetricSliceTest.KEY_REQ));
+                    }
+                    if (data.containsKey(TrafficMetricSliceTest.KEY_RSP)) {
+                        response.set(data.get(TrafficMetricSliceTest.KEY_RSP));
+                    }
```

@g4s8 it seems that this code block is not needed and so may be removed. It would be even better to throw an exception here, because we do not expect the slice to update any gauges.